libs/books/data-access/src/lib/+state/reading-list.reducer.ts

```diff
@@ -58,7 +63,27 @@ const readingListReducer = createReducer(
   ),
   on(ReadingListActions.removeFromReadingListFailed, (state, action) =>
     readingListAdapter.addOne(action.item, state)
-  )
+  ),
+  on(ReadingListActions.markBookAsFinished, (state, action) => {
+    const updatedItem: Update<ReadingListItem> = {
+      id: action.bookId,
+      changes: {
+        finished: true,
+        finishedDate: action.finishedDate
+      }
+    };
+    return readingListAdapter.updateOne(updatedItem, state);
+  }),
+  on(ReadingListActions.markBookAsFinishFailed, (state, action) => {
+    const updatedItem: Update<ReadingListItem> = {
+      id: action.bookId,
+      changes: {
+        finished: false,
+        finishedDate: ''
```

Comment:
We could get rid of the `finished` attribute overall, and make `finishedDate` undefined for not finished books.